### PR TITLE
Dev refactor generic adder  tests to use base class

### PR DESCRIPTION
**Pull request by ekneg54**

### CODE REVIEW

**Comment by ppcad** [comment]
The changes resulted in some unused imports in:
* generic_adder/processor.py
* test_generic_adder.py

Could you please remove them?

**Comment by ekneg54** [comment]
done

**Review by ppcad — APPROVED** [review]

**Comment by ekneg54** [comment]
rebased from main

**Comment by ekneg54** [comment]
rebased from new main

**Review by urdit52 — APPROVED** [review]
looks good to me. I like the parameterised version of tests as proven in the tests for rule.py.

File: logprep/processor/generic_adder/factory.py [hunk 1]
```diff
@@ -7,16 +7,16 @@
 class GenericAdderFactory(BaseFactory):
     """Create generic adder."""
 
+    mandatory_fields = ["generic_rules", "specific_rules"]
```
[the remark this answers is not in the document]
Comment (ekneg54) [reply logprep/processor/generic_adder/factory.py:10]: is proofed in parent

[the remark this answers is not in the document]
Comment (urdit52) [reply logprep/processor/generic_adder/factory.py:10]: good.

**Pull request merged**